### PR TITLE
docs(demo): add demo page 

### PR DESCRIPTION
**Pull request by 317brian**
Resolves issue: [DOC-317]

[DOC-123]: https://armory.atlassian.net/browse/DOC-123
[DOC-317]: https://armory.atlassian.net/browse/DOC-317

### CODE REVIEW

File: content/en/docs/Overview/demo-armory.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+## Requirements
+
+Here are the requirements for completing this demo:
+
+* You need to be able to provision instances in your cloud environment.
+* The provisioned instance needs to meet the following minimum requirements
+  * Ubuntu 18.04
+  * 4 vCPUs
+  * 16 GB of memory
```
Comment (317brian) [inline content/en/docs/Overview/demo-armory.md]: extracted these from t3.xlarge specs

File: content/en/docs/Overview/demo/_index.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+- Infrastructure as Code (IaC) using gitops and Terraform
+- Policy Driven Deployment (PDD) - for security and compliance (Security as Code)
+- Declarative Spinnaker Application and Pipelines for (Pipelines as Code)
+- Automatic EKS Provisioning through gitops
+- Armory Agent for Kubernetes (delegate for kubernetes)
+- Enterprise Armory Platform powered by Spinnaker
+- Built in Kubernetes deployment namespace for testing (QA, Stage, Prod)
+- Extension points through yaml (Jenkins, Git, JIRA, Docker hub)
+
+Minnaker uses [K3s](https://k3s.io/) as the kubernetes server. If you already have a kubernetes cluster for testing you can skip to [Install Minnaker](/docs/overview/demo/install-demo#install-minnaker).
```
Comment (aimeeu) [inline content/en/docs/Overview/demo/_index.md]:
```suggestion
Minnaker uses [K3s](https://k3s.io/) as the Kubernetes server. If you already have a Kubernetes cluster for testing, you can skip to [Install Minnaker]({{< ref "install-demo#install-minnaker" >}}).
```

File: content/en/docs/Overview/demo/install-demo.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    ```bash
+    # Download the latest release of Minnaker:
+    curl -LO https://github.com/armory/minnaker/releases/latest/download/minnaker.tgz
+    # Extract Minnaker from the archive:
+    tar -xzvf minnaker.tgz
+    # Execute the installation script:
+    ./minnaker/scripts/install.sh
+    ```
+
+    The commands above declaratively deploy Armory using git clone, [kustomize](https://kustomize.io/), and [Armory Operator](/docs/installation/operator/).  The entire configuration is gitops enabled for you so you can ‘git push’ your configurations to your own source control.
```
Comment (aimeeu) [inline content/en/docs/Overview/demo/install-demo.md]:
```suggestion
    The commands above declaratively deploy Armory using `git clone`, [Kustomize](https://kustomize.io/), and [Armory Operator]({{< ref "operator" >}}).  The entire configuration is GitOps enabled for you so you can ‘git push’ your configurations to your own source control. 
```

**Review by aimeeu — CHANGES_REQUESTED** [review]

**Comment by alexfornuto** [comment]
The PR is now dependent on #449, as it links to pages created there. Once #449 is merged I will rebase this PR and submit it for review.

File: content/en/docs/Overview/demo/_index.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+linkTitle: "Armory Demo"
+no_list: true
+description: "This demo is a fully featured proof of concept (PoC) of Armory Enterprise for Spinnaker™. Walk through installation, configuration, and deployment using sample applications."
+aliases:
+  - /spinnaker-install-admin-guides/architecture/
+---
+
+## Overview
+
+Armory Enteprise for Spinnaker is an superset of Spinnaker, which itself is a collection of microservices working in concert to create a single continuous delivery / deployment service.
```
Comment (dogonthehorizon) [inline content/en/docs/Overview/demo/_index.md]:
```suggestion
Armory Enteprise for Spinnaker is a superset of Spinnaker, which itself is a collection of microservices working in concert to create a single continuous delivery / deployment service.
```

\<nitpick /\>

File: content/en/docs/Overview/demo/_index.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+Installation and configuration of Armory Enterprise for Spinnaker can seem a daunting task for those not yet familiar with open-source Spinnaker, so we created Minnaker: a demo deployment of Spinnaker designed to run in a single instance (EC2 or other virtual machine).
+
+## Features
+
+What is included in Armory Easy Poc?
```
Comment (dogonthehorizon) [inline content/en/docs/Overview/demo/_index.md]: > Armory Easy Poc?

Should this be Minnaker?

File: content/en/docs/Overview/demo/_index.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+Installation and configuration of Armory Enterprise for Spinnaker can seem a daunting task for those not yet familiar with open-source Spinnaker, so we created Minnaker: a demo deployment of Spinnaker designed to run in a single instance (EC2 or other virtual machine).
+
+## Features
+
+What is included in Armory Easy Poc? 
+
+Armory Enterprise 
+
+- Infrastructure as Code (IaC) using gitops and Terraform
```
Comment (dogonthehorizon) [inline content/en/docs/Overview/demo/_index.md]: `leveraging` perhaps conveys better intent here. `gitops` isn't a tool so much as its a certain way of using IaC manifests in some kind of deployment

File: content/en/docs/Overview/demo/_index.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+- Policy Driven Deployment (PDD) - for security and compliance (Security as Code)
+- Declarative Spinnaker Application and Pipelines for (Pipelines as Code)
```
Comment (dogonthehorizon) [inline content/en/docs/Overview/demo/_index.md]: Should Application be plural as well?

File: content/en/docs/Overview/demo/_index.md [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+- Automatic EKS Provisioning through gitops
```
Comment (dogonthehorizon) [inline content/en/docs/Overview/demo/_index.md]: What does this mean exactly, does the demo provision EKS clusters?

File: content/en/docs/Overview/demo/_index.md [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+- Armory Agent for Kubernetes (delegate for kubernetes)
```
Comment (dogonthehorizon) [inline content/en/docs/Overview/demo/_index.md]: Should we link to agent documentation here?

File: content/en/docs/Overview/demo/_index.md [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+- Enterprise Armory Platform powered by Spinnaker
```
Comment (dogonthehorizon) [inline content/en/docs/Overview/demo/_index.md]: Could you help me understand what is meant with this feature?

File: content/en/docs/Overview/demo/_index.md [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+- Built in Kubernetes deployment namespace for testing (QA, Stage, Prod)
+- Extension points through yaml (Jenkins, Git, JIRA, Docker hub)
+
+Minnaker uses [K3s](https://k3s.io/) as the Kubernetes server. If you already have a Kubernetes cluster for testing, you can skip to [Install Minnaker]({{< ref "install-demo#install-minnaker" >}}).
```
Comment (dogonthehorizon) [inline content/en/docs/Overview/demo/_index.md]:
```suggestion
Minnaker uses [K3s](https://k3s.io/) as the Kubernetes implementation. If you already have a Kubernetes cluster for testing, you can skip to [Install Minnaker]({{< ref "install-demo#install-minnaker" >}}).
```

File: content/en/docs/Overview/demo/_index.md [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+Here are the requirements for completing this demo:
+
+* You need to be able to provision instances in your cloud environment.
+* The provisioned instance needs to meet the following minimum requirements
+  * Ubuntu 18.04
+  * 4 vCPUs
+  * 16 GB of memory
+  * 50 GB of storage
+
+If deploying to an EC2 instance, we suggest using the t2.xlarge tier.
```
Comment (dogonthehorizon) [inline content/en/docs/Overview/demo/_index.md]: If we're suggesting a default instance size can we make it `t3a.xlarge`? It is the newest and cheapest of the `t` offerings from Amazon, and [comparable in traditional specs](https://aws.amazon.com/ec2/pricing/on-demand/) to the `t2` class instances.

Comment (dogonthehorizon) [inline content/en/docs/Overview/demo/_index.md]:
```suggestion
If deploying to an EC2 instance, we suggest using a t3a.xlarge instance.
```

File: content/en/docs/Overview/demo/_index.md [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+
+> You don't need to create this instance ahead of following the guide, only confirm access to the resources required.
+
+### Concepts
+
+The Minnaker demo uses the Armory Operator to manage services, and Kustomize to manage configuration files. If you're unfamiliar with either, check out the following docs before you being:
```
Comment (dogonthehorizon) [inline content/en/docs/Overview/demo/_index.md]:
```suggestion
The Minnaker demo uses the Armory Operator to manage services, and Kustomize to manage configuration files. If you're unfamiliar with either, check out the following docs before you begin:
```

File: content/en/docs/Overview/demo/configure-integrations.md [hunk 14]
```diff
@@ -0,0 +1,182 @@
...
+
+This guide assumes you have an implementation of or access to these systems. The demo does not provide a Jenkins instance, for example.
+
+## Jenkins
+
+These steps will let Armory Spinnaker use a Jenkins build as a trigger for deployment. 
+
+### Set variables
+
+1. In your Jenkins instance, create a new user and API token to authenticate to your Jenkins with. The Token will be stored as a Kubernetes secret, The user and host name will be stored in a yaml patch file to be added to `SpinnakerService.yaml`, which will enable the integration.
```
Comment (dogonthehorizon) [inline content/en/docs/Overview/demo/configure-integrations.md:25]: Should we link to docs on how to create api users?

File: content/en/docs/Overview/demo/configure-integrations.md [hunk 15]
```diff
@@ -0,0 +1,182 @@
...
+
+1. From the Minnaker host CLI, temporarily set the user, token, and host name as envirnment variables:
```
Comment (dogonthehorizon) [inline content/en/docs/Overview/demo/configure-integrations.md]:
```suggestion
1. From the Minnaker host CLI, temporarily set the user, token, and host name as environment variables:
```

File: content/en/docs/Overview/demo/configure-integrations.md [hunk 16]
```diff
@@ -0,0 +1,182 @@
...
+    {{< prism lang=yaml line=8 >}}
+      # Artifacts & Accounts
+      - accounts/kubernetes/patch-kube.yml      # Kubernetes accounts
+      - accounts/docker/patch-dockerhub.yml     # Docker accounts
+      - accounts/http/patch-http.yml            # HTTP accounts
+      - accounts/git/patch-github.yml           # Public GitHub Repos
+      - accounts/git/patch-gitrepo.yml          # Public Git Repos
+      # - accounts/notifications/patch-slack.yml # Slack Notifications
+      - accounts/ci/patch-jenkins.yml          # Jenkins CI Integration
+      # - accounts/canary/prometheus.yml        # TODO - need to add CONFIG STORE
```
Comment (dogonthehorizon) [inline content/en/docs/Overview/demo/configure-integrations.md:74]: Do we want to leave this here?

File: content/en/docs/Overview/demo/configure-integrations.md [hunk 17]
```diff
@@ -0,0 +1,182 @@
...
+1. Create a [Personal Access Token](https://github.com/settings/tokens) in your GitHub account to allow Armory to authenticate to your GitHub repos. The Token will be stored as a Kubernetes secret.
+
+1. From your Minnaker host, edit the file `spinsvc/secrets/secrets-example.env` to apply your token to the `github-token` key. You can also use `sed`:
+
+    ```bash
+    cd ~/minnaker
+    sed -i -e 's/github-token=xxx/github-token=<github-token>/g' spinsvc/secrets/secrets-example.env
+    ```
+
+1. Run `create-secrets.sh` in `spinsvc/secrets` to update the [Kubernetes Secrets](https://kubernetes.io/docs/concepts/configuration/secret/) with the new token.
```
Comment (dogonthehorizon) [inline content/en/docs/Overview/demo/configure-integrations.md]:
```suggestion
1. Run `create-secrets.sh` in `spinsvc/secrets` to update the generated [Kubernetes Secret](https://kubernetes.io/docs/concepts/configuration/secret/) with the new token.
```

File: content/en/docs/Overview/demo/configure-integrations.md [hunk 18]
```diff
@@ -0,0 +1,182 @@
...
+                token: encrypted:k8s!n:spin-secrets!k:github-token             # (Secret). GitHub # token.
+    ```
+
+1. Invoke `deploy.sh` to apply the changes:
+
+    ```bash
+    ./spinsvc/deploy.sh
+    ```
+
+   You can use watch kubectl get pods,spinsvc -n spinnaker from the host shell to monitor the status of the redeployment.
```
Comment (dogonthehorizon) [inline content/en/docs/Overview/demo/configure-integrations.md]:
```suggestion
   You can use `watch kubectl get pods,spinsvc -n spinnaker` from the host shell to monitor the status of the redeployment.
```

To be consistent with other usages of this line

File: content/en/docs/Overview/demo/install-demo.md [hunk 19]
```diff
@@ -0,0 +1,68 @@
...
+    tar -xzvf v0.1.2.tar.gz
+    # Rename the minnaker directory:
+    mv minnaker-0.1.* minnaker
+    # Execute the installation script:
+    ./minnaker/scripts/install.sh
+    ```
+
+    The commands above declaratively deploy Armory using `git clone`, [Kustomize](https://kustomize.io/), and [Armory Operator]({{< ref "operator" >}}).  The entire configuration is GitOps enabled for you so you can ‘git push’ your configurations to your own source control. 
+
+1. The creating of namespaces, loading for CRD and Operator takes around 3 minutes to get Spinnaker fully installed with all services in a “Running” state. The script ends by running a `watch` command to monitor the state of deployment. The environment is ready when the `spinnakerservice` service reads an "OK" status:
```
Comment (dogonthehorizon) [inline content/en/docs/Overview/demo/install-demo.md]:
```suggestion
1. The creating of namespaces, loading for CRD and Operator takes around 3 minutes to get Spinnaker fully installed with all services in a “Running” state. The script ends by running a `watch` command to monitor the state of deployment. The environment is ready when the `spinnakerservice` object reads an "OK" status:
```

File: content/en/docs/Overview/demo/_index.md (hunk 4 above)
Comment (alexfornuto) [reply content/en/docs/Overview/demo/_index.md]: Not a nitpick, a legitimate copy edit. +1

File: content/en/docs/Overview/demo/_index.md (hunk 5 above)
Comment (alexfornuto) [reply content/en/docs/Overview/demo/_index.md]: Probably.

File: content/en/docs/Overview/demo/_index.md (hunk 8 above)
Comment (alexfornuto) [reply content/en/docs/Overview/demo/_index.md]: I think this whole section is more about what Spinnaker and/or Armory Enterprise can do, not what this Minnaker demo provides. What do you think, should it be edited to be specific to what this guide demonstrates, or to explicitly be about what we _can_ offer?

CC @armory/docs-team for comment.

File: content/en/docs/Overview/demo/configure-integrations.md (hunk 16 above)
Comment (alexfornuto) [reply content/en/docs/Overview/demo/configure-integrations.md:74]: That's what's in the file when `kustomization.yml` is installed via the Minnaker script. The docs should reflect what the user sees, so if we want to remove this it should be from Minnaker first.

File: content/en/docs/Overview/demo/configure-integrations.md (hunk 18 above)
Comment (alexfornuto) [reply content/en/docs/Overview/demo/configure-integrations.md]: After moving it to include `watch`, big :+1:

File: content/en/docs/Overview/demo/configure-integrations.md (hunk 14 above)
Comment (alexfornuto) [reply content/en/docs/Overview/demo/configure-integrations.md:25]: If I could find a good doc provided by Jenkins on this, I'd link to it. [This](https://www.jenkins.io/doc/book/system-administration/authenticating-scripted-clients/) is the closest I found, and I'm not sure it's useful for this application.

**Pull request closed without merging**